metaflow/task.py

```diff
@@ -188,10 +197,10 @@ def lineage():
                     if join_type == "foreach":
                         top = i["_foreach_stack"][-1]
                         bottom = i["_foreach_stack"][:-1]
-                        # the topmost indices in the stack are all
-                        # different naturally, so ignore them in the
+                        # the topmost indices and values in the stack are
```

Comment:
nit: I know this is not in your modification but what assert?

Reply:
I think its referring to the `not all_equal(lineage())` below.